src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -1125,10 +1134,29 @@ trait ChoosesMode {
           if (isFirstOrLastTrip) WALK_BASED.allowedBeamModesForFirstAndLastLeg
           else WALK_BASED.allowedBeamModes
         walkBasedModes ++ enabledModes
+      case Some(BIKE_BASED) =>
+        if (BIKE_BASED.vehicleSharedOrNot(availablePersonalStreetVehicles)) {Seq[BeamMode]()}
+        else BIKE_BASED.allowedBeamModes
       case Some(tourMode) => tourMode.allowedBeamModes
```

Comment:
With the change below in `Modes` we could just replace this line with

      case Some(tourMode) => tourMode.allowedBeamModesGivenAvailableVehicles(availablePersonalStreetVehicles, isFirstOrLastTrip)